app/views/NotificationPreferencesView/index.tsx

```diff
@@ -1,282 +1,234 @@
+import { CustomIcon } from '../../containers/CustomIcon';
 import * as List from '../../containers/List';
-import I18n from '../../i18n';
-import { TSupportedThemes, withTheme } from '../../theme';
-import protectedFunction from '../../lib/methods/helpers/protectedFunction';
 import SafeAreaView from '../../containers/SafeAreaView';
+import StatusBar from '../../containers/StatusBar';
+import { IRoomNotifications, TRoomNotificationsModel } from '../../definitions';
+import I18n from '../../i18n';
+import { SWITCH_TRACK_COLOR } from '../../lib/constants';
+import { useAppSelector } from '../../lib/hooks';
+import { showErrorAlertWithEMessage } from '../../lib/methods/helpers';
+import { compareServerVersion } from '../../lib/methods/helpers/compareServerVersion';
 import log, { events, logEvent } from '../../lib/methods/helpers/log';
+import { Services } from '../../lib/services';
+import { ChatsStackParamList } from '../../stacks/types';
+import { useTheme } from '../../theme';
 import sharedStyles from '../Styles';
 import { IOptionsField, OPTIONS } from './options';
-import { ChatsStackParamList } from '../../stacks/types';
-import { IApplicationState, IRoomNotifications, TRoomNotificationsModel } from '../../definitions';
-import { Services } from '../../lib/services';
-import { compareServerVersion } from '../../lib/methods/helpers/compareServerVersion';
-
-const styles = StyleSheet.create({
-	pickerText: {
-		...sharedStyles.textRegular,
-		fontSize: 16
-	}
-});
 
-interface INotificationPreferencesViewProps {
-	navigation: StackNavigationProp<ChatsStackParamList, 'NotificationPrefView'>;
-	route: RouteProp<ChatsStackParamList, 'NotificationPrefView'>;
-	theme: TSupportedThemes;
-	serverVersion: string | null;
-}
-
-interface INotificationPreferencesViewState {
+interface IBaseParams {
+	preference: string;
 	room: TRoomNotificationsModel;
+	onChangeValue: (pref: string, param: { [key: string]: string }, onError: () => void) => void;
 }
 
-class NotificationPreferencesView extends React.Component<INotificationPreferencesViewProps, INotificationPreferencesViewState> {
-	static navigationOptions = () => ({
-		title: I18n.t('Notification_Preferences')
-	});
-
-	private mounted: boolean;
-	private rid: string;
-	private roomObservable?: Observable<TRoomNotificationsModel>;
-	private subscription?: Subscription;
-
-	constructor(props: INotificationPreferencesViewProps) {
-		super(props);
-		this.mounted = false;
-		this.rid = props.route.params?.rid ?? '';
-		const room = props.route.params?.room;
-		this.state = {
-			room: room || {}
-		};
-		if (room && room.observe) {
-			this.roomObservable = room.observe();
-			this.subscription = this.roomObservable.subscribe(changes => {
-				if (this.mounted) {
-					this.setState({ room: changes });
-				} else {
-					// @ts-ignore
-					this.state.room = changes;
-				}
-			});
-		}
-	}
-
-	componentDidMount() {
-		this.mounted = true;
-	}
+const RenderListPicker = React.memo(
```

Comment:
`room` is an object and `onChangeValue` is a function, so React.memo isn't really doing anything.